daft/dataframe/dataframe.py

```diff
@@ -1691,6 +1691,63 @@ def write_bigtable(
 
         return df_to_write.write_sink(sink)
 
+    @DataframePublicAPI
+    def write_sql(
+        self,
+        table_name: str,
+        conn: Union[str, Callable[[], Any]],
+        *,
+        mode: Literal["append", "overwrite"],  # NOTE: keep this aligned with daft.io.sql.SQL_SINK_MODES.
+        # NOTE:                                        Must do so manually because Python doesn't support circular imports.
+    ) -> "DataFrame":
+        """Write a DataFrame into a relational database table.
+
+        Supports PostgreSQL, SQLite, MySQL, and other SQLAlchemy-compatible databases.
+
+        Args:
+            table_name: Name of the target SQL table to write to.
+            conn: Connection string (URL) or a callable that returns a SQLAlchemy connection.
+                Examples of connection strings:
+                - PostgreSQL: "postgresql://user:password@localhost:5432/mydb"
+                - SQLite: "sqlite:///path/to/database.db"
+                - MySQL: "mysql+pymysql://user:password@localhost:3306/mydb"
+            mode: Write mode for the operation. One of:
+                - "append": Append data to an existing table.
+                - "overwrite": Drop the table, if it exists, and create a new table with new data.
+
+        Returns:
+            DataFrame: A new DataFrame containing metadata about the write operation
+                (rows_written, bytes_written, table_name).
+
+        Note:
+            This call is **blocking** and will execute the DataFrame when called.
```

Comment:
Remove Note, add Warn that this is experimental and currently does not work with mysql

Reply:
added `Supports PostgreSQL and SQLite databases. This is still an experimental feature. Support for additional databases is ongoing.` and removed the mysql connection string example